### PR TITLE
gha: Manually rebase PR atop of the target branch before testing

### PR DESCRIPTION
**Pull request by fidencio**
We're changing what's been done as part of ac939c458cde, as we've notcied issues using `github.event.pull_request.merge_commit_sha`.

Basically, whenever a force-push would happen, the reference of merge_commit_sha wouldn't be updated, leading us to test PRs with the old code. :-/

In order to get the rebase properly working, we need to ensure we pull the hash of the commit as part of checkout action, and ensure fetch-depth is set to 0.

Fixes: #7414

### CODE REVIEW

File: .github/workflows/ci-on-push.yaml [hunk 1]
```diff
@@ -28,4 +28,5 @@ jobs:
       commit-hash: ${{ github.event.pull_request.merge_commit_sha }}
```
Comment (stevenhorsman) [inline .github/workflows/ci-on-push.yaml:28]: Do we want to change this back to `commit-hash: ${{ github.event.pull_request.head.sha }}` now we are rebasing ourselves?

Comment (fidencio) [reply .github/workflows/ci-on-push.yaml:28]: We do! :-)

Comment (fidencio) [reply .github/workflows/ci-on-push.yaml]: Fixed!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks

**Review by GabyCT — APPROVED** [review]

**Pull request merged**